pathfinder/tests/test_mocked_integration.py

```diff
@@ -0,0 +1,335 @@
+# -*- coding: utf-8 -*-
+from typing import List
+
+from unittest.mock import Mock
+from web3 import Web3
+from raiden_contracts.contract_manager import ContractManager
+from raiden_libs.test.mocks.blockchain import BlockchainListenerMock
+
+from pathfinder.token_network import TokenNetwork
+from pathfinder.pathfinding_service import PathfindingService
+from pathfinder.utils.types import Address
+
+
+def test_pathfinding_service_with_mocked_events(
+    token_networks: List[TokenNetwork],  # just used for addresses
+    addresses: List[Address],
+    pathfinding_service_mocked_listeners: PathfindingService
+):
+    network_listener = pathfinding_service_mocked_listeners.token_network_listener
+    registry_listener = pathfinding_service_mocked_listeners.token_network_registry_listener
+
+    token_network_address = token_networks[0].address
+
+    # this is a new Pathfinding Service, there should be no token networks registered
+    assert len(pathfinding_service_mocked_listeners.token_networks.keys()) == 0
+
+    # emit a TokenNetworkCreated event
+    registry_listener.emit_event(dict(
+        name='TokenNetworkCreated',
+        args=dict(
+            token_network_address=token_network_address
+        )
+    ))
+
+    # now there should be a token network registered
+    assert token_network_address in pathfinding_service_mocked_listeners.token_networks
+    token_network = pathfinding_service_mocked_listeners.token_networks[token_network_address]
+
+    # Now initialize some channels in this network.
+    # The tuples in channel_descriptions define the following:
+    # (
+    #     p1_index,
+    #     p1_deposit,
+    #     p1_transferred_amount,
+    #     p1_fee,
+    #     p2_index,
+    #     p2_deposit,
+    #     p2_transferred_amount,
+    #     p2_fee
+    # )
+    # Topology:
+    #       /-------------\
+    # 0 -- 1 -- 2 -- 3 -- 4    5 -- 6
+    #  \-------/
+
```

Comment:
The fees are not used here. I'd either leave them out or mock the fee functionality to make the test more integrational.

Reply:
I'd keep them for the future and integrate this as soon as the message format is known.